spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/config/GcpConfigProperties.java

```diff
@@ -130,4 +116,20 @@ public Credentials getCredentials() {
 		return this.credentials;
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		if (this.profile == null) {
+			String[] profiles = applicationContext.getEnvironment().getActiveProfiles();
```

Comment:
Should we add `Assert.notNull()` on `applicationContext` and` applicationContext.getEnvironment()`, to simplify troubleshooting in the future?

Reply:
I believe the contract of `ApplicationContextAware` is that it provides a non-null context, and the context should always have an environment.

Reply:
That's correct.
Although I think we can narrow the responsibility to the `EnvironmentAware`  hook: nor reason to deal with the whole `ApplicationContext` here.